test/unit/org/apache/cassandra/batchlog/BatchlogEndpointFilterTest.java

```diff
@@ -56,7 +87,25 @@ public void shouldUseLocalRackIfPreferred() throws UnknownHostException
                                                                           .put("2", InetAddressAndPort.getByName("2"))
                                                                           .put("2", InetAddressAndPort.getByName("22"))
                                                                           .build();
-        Collection<InetAddressAndPort> result = filterBatchlogEndpoints(true, endpoints);
+        Collection<InetAddressAndPort> result = filterBatchlogEndpointsRandom(true, endpoints);
+        assertThat(result.size()).isEqualTo(2);
+        assertThat(result).containsAnyElementsOf(endpoints.get(LOCAL));
+        assertThat(result).containsAnyElementsOf(Iterables.concat(endpoints.get("1"), endpoints.get("2")));
+    }
+
+    @Test
+    public void shouldUseLocalRackIfPreferredStrategy() throws UnknownHostException
```

Comment:
how is this test different than `shouldUseLocalRackIfPreferred()` ? I do see there are two different batchlogEndpointStrategies but I think the name of the tests are the same and they don't express the difference in tests 🤔